### ISSUE TITLE
login: can't change directory to '/home/tsuser'

### ISSUE THREAD

**Issue by jens-maus** [Thinstation/thinstation#127]
The current build system seems to be largely dependent on the current read/write permissions the files are having when creating/building an image. E.g. if a user has a very restrictive umask setting (e.g. 0007) and he is checking out the thinstation github repository directory he might end up with a non-working thinstation image which at the time something is automatically started via `agetty -a tsuser` returns and
error message like

```
login: can't change directory to '/home/tsuser'
```

on console 1.

running a chmod over the whole thinstation build environment before running setup-chroot to build an image resolves this problem. Thus, the build environment should be changed to actually ensure that no matter which permissions files/directories have in the build environment a running thinstation image should be generated with relaxed permissions.

**Comment by Doncuppjr** [comment]
That's a sticky situation. Not every file needs the same perms, so a recursive chmod would still break stuff, like suid bins, and allow access to files that people should not have access to.
Sent from my Verizon Wireless 4G LTE smartphone

-------- Original message --------

**Comment by jens-maus** [comment]
Well, here I did the following right before running setup-chroot to fix the problems I had:

```
$ find . -executable -exec chmod a+rx {} \;
$ find . -type f -exec chmod a+r {} \;
```

This doesn't touch the suid stuff and will simply assure that there are proper permissions set. I also made sure that the `umask` before running setup-chroot is set up 022. 

However, I still feel that this could be directly fixed in thinstation (at least be made optional) using the following things:

1. modify `ts/build/build` to have a function like `fix_permission()` similar to `strip_files()` which will run through the `tmp-tree` right before creating the image, making sure permissions are properly set like above.
2. modify setup-chroot to set a proper `umask` itself to assure that newly created files appear with the right permissions.

As said, this could be made optional because there might be some environment (like here) where assuring that the build ends up in a running thinstation environment is more important than assuring that permissions are set exactly like the files are stored in git.
